### PR TITLE
Added tests for XMLProfileManager [10089]

### PR DESCRIPTION
**Pull request by imontesino**
Merge after #1612

Added unit tests for the `XMLProfileManager` classes' public methods

### CODE REVIEW

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/3219/)
* Linux aarch64[Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux%20aarch64/430/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/2884/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/3251/)

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/3222/)
* Linux aarch64[Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux%20aarch64/433/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/2886/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/3254/)

**Review by rsanchez15 — CHANGES_REQUESTED** [review]
Apply also all format suggestions from the review of #1605

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/3274/)
* Linux aarch64[Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux%20aarch64/485/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/2925/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/3306/)

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/3387/)
* Linux aarch64[Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux%20aarch64/585/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/3037/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/3419/)

File: src/cpp/rtps/xmlparser/XMLProfileManager.cpp [hunk 1]
```diff
@@ -324,71 +315,22 @@ XMLP_ret XMLProfileManager::loadXMLFile(
...
-            {
-                return loaded_ret;
-            }
-        }
-    }
-
-    return XMLP_ret::XML_ERROR;
-}
-
-XMLP_ret XMLProfileManager::extractDynamicTypes(
```
Comment (rsanchez15) [inline src/cpp/rtps/xmlparser/XMLProfileManager.cpp:343]: `extractDynamicTypes()` is still declared in `XMLProfileManager.h` but I can't find its definition.

File: test/unittest/xmlparser/XMLProfileParserTests.cpp [hunk 2]
```diff
@@ -105,6 +105,105 @@ TEST_F(XMLProfileParserTests, XMLoadProfiles)
...
+            EXPECT_EQ(xmlparser::XMLP_ret::XML_ERROR, xmlparser::XMLProfileManager::loadXMLProfiles(*titleElement));
+            xmlparser::XMLProfileManager::DeleteInstance();
+        }
+    }
+}
+
+/*
+ * This test checks the return of the loadXMLDynamicTypes method
+ * 1. Check correct return
+ * 1. Check error return
```
Comment (rsanchez15) [inline test/unittest/xmlparser/XMLProfileParserTests.cpp:164]:
```suggestion
 * 2. Check error return
```

File: test/unittest/xmlparser/XMLProfileParserTests.cpp [hunk 3]
```diff
@@ -872,94 +995,286 @@ TEST_F(XMLProfileParserTests, SHM_transport_descriptors_config)
...
+    EXPECT_EQ(nullptr, xmlparser::XMLProfileManager::getTransportById("wrong_type"));
+}
 
+/*
+ * Tests whether the extraction of XML profiles succeeds when all profiles are correct.
+ * XMLProfileManager::loadXMLNode returns XMLProfileManager::extractProfiles.
+ * The following cases are tested:
+ *  1. A corrrect standalone profile element is parsed, return value of XMLP_ret::XML_OK is expected
+ *  1. A corrrect rooted profile element is parsed, return value of XMLP_ret::XML_OK is expected
+ *  1. An incorrect profile element is parsed, return value of XMLP_ret::XML_ERROR is expected
```
Comment (rsanchez15) [inline test/unittest/xmlparser/XMLProfileParserTests.cpp:1031]:
```suggestion
 *  2. A corrrect rooted profile element is parsed, return value of XMLP_ret::XML_OK is expected
 *  3. An incorrect profile element is parsed, return value of XMLP_ret::XML_ERROR is expected
```

File: test/unittest/xmlparser/XMLProfileParserTests.cpp [hunk 4]
```diff
@@ -872,94 +995,286 @@ TEST_F(XMLProfileParserTests, SHM_transport_descriptors_config)
...
+        // same profile name twice
+        sprintf(xml, xml_p, "test_name");
+        ASSERT_EQ(tinyxml2::XMLError::XML_SUCCESS, xml_doc.Parse(xml));
+        EXPECT_EQ(xmlparser::XMLP_ret::XML_OK, xmlparser::XMLProfileManager::loadXMLNode(xml_doc));
+        EXPECT_EQ(xmlparser::XMLP_ret::XML_ERROR, xmlparser::XMLProfileManager::loadXMLNode(xml_doc));
+
+        xmlparser::XMLProfileManager::DeleteInstance();
+    }
+
+    {
```
Comment (rsanchez15) [inline test/unittest/xmlparser/XMLProfileParserTests.cpp:1206]: Add a brief comment in each block explaining which case you are testing.

File: test/unittest/xmlparser/XMLProfileParserTests.cpp [hunk 5]
```diff
@@ -998,6 +1313,156 @@ TEST_F(XMLProfileParserTests, skip_default_xml)
...
+ *  4. On giving an empty string as a filename, expect XMLP_ret::XML_ERROR.
+ *  5. On parsing the same file twice, expect XMLP_ret::XML_OK.
+ *  6. On parsing an incorrect element inside profiles element, expect XMLP_ret::XML_ERROR.
+ */
+TEST_F(XMLProfileParserTests, loadXMLFile)
+{
+    const char* filename = "minimal_file.xml";
+    tinyxml2::XMLDocument xml_doc;
+
+    {
```
Comment (rsanchez15) [inline test/unittest/xmlparser/XMLProfileParserTests.cpp:1367]: Same as before. Brief comment for each block. You can copy and paste the comments above in its corresponding block

File: include/fastrtps/xmlparser/XMLProfileManager.h [hunk 6]
```diff
@@ -222,6 +222,7 @@ class XMLProfileManager
         subscriber_profiles_.clear();
```
Comment (rsanchez15) [inline include/fastrtps/xmlparser/XMLProfileManager.h:222]: Missing negative test for `CreateDynamicPubSubType()` (line 242).

**Review by rsanchez15 — CHANGES_REQUESTED** [review]

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/3526/)
* Linux aarch64[Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux%20aarch64/716/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/3134/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/3558/)

**Review by rsanchez15 — APPROVED** [review]

**Pull request merged**